### PR TITLE
[eas-cli] ensure that all links are clickable, unify display

### PR DESCRIPTION
**Pull request by Simek**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

We have an util to format links in messages, but it is not used consistently.

# How

Ensure that all links included in messages are decorated with `link` util, relay on `fallback` instead of directly checking the flag.

# Test Plan

All `eas-cli` tests are passing.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,8 @@ This is the log of notable changes to EAS CLI and related packages.
 
 ### 🧹 Chores
 
+- Make sure that all links in messages are clickable, if used terminal supports that. ([#1591](https://github.com/expo/eas-cli/pull/1591) by [@Simek](https://github.com/Simek))
```
Comment (dsokal) [inline CHANGELOG.md:17]: Maybe this is me but this sentence doesn't make sense.

File: packages/eas-cli/src/build/build.ts [hunk 2]
```diff
@@ -362,7 +362,7 @@ async function handleSingleBuildProgressAsync(
       spinner.succeed('Build finished');
       return { refetch: false };
     case BuildStatus.New:
-      spinner.text = `Build is waiting to enter the queue. Check your concurrency limit at ${chalk.underline(
+      spinner.text = `Build is waiting to enter the queue. Check your concurrency limit at ${link(
         formatAccountSubscriptionsUrl(accountName)
       )}.`;
```
Comment (dsokal) [inline packages/eas-cli/src/build/build.ts:367]: Are you sure this works in spinners?

**Review by dsokal — APPROVED** [review]

**Review by wkozyra95 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Simek) [reply CHANGELOG.md:17]: Do you have any suggestions how to improve it? Which part seems off for you, or is it the whole sentence?

> Ensure that all URLs in messages are clickable, if used terminal supports mouse interaction. 

Does the entry above makes more sense?

Comment (dsokal) [reply CHANGELOG.md:17]: "Make all URLs in logs clickable in terminals supporting hyperlinks."?

Comment (Simek) [reply CHANGELOG.md:17]: Want to suggest that change, so we can update it from web UI?

Comment (dsokal) [reply CHANGELOG.md:17]:
```suggestion
- Make all URLs in logs clickable in terminals supporting hyperlinks. ([#1591](https://github.com/expo/eas-cli/pull/1591) by [@Simek](https://github.com/Simek))
```

**Pull request merged**